packages/notifications-service/src/resolver.ts

```diff
@@ -0,0 +1,114 @@
+              return notificationConfig;
+            } );
+        } ).catch( err => {
+          throw err;
+        } );
+    },
+    getNotificationsConfigBy ( root: any, { selectors }: GraphQLArgs, ctx: any ) {
+      return NotificationsConfig
+        .find( selectors )
+        .lean<NotificationsConfigType>()
+        .exec()
+        .then( async notificationConfigs => {
+          /* Constructing getHomeTypeBy queries */
+          const queries = notificationConfigs
+            .reduce<string[]>( ( acc, notificationConfig ) => {
+              if ( acc.includes( notificationConfig.source ) ) {
+                return acc;
+              }
+              return [
+                ...acc,
+                notificationConfig.source,
+              ];
+            }, [] )
+            .map( ( source, index ) => {
+              return `source_${ index }: getHomeTypeBy(input: { _id: "${ source }" }) { _id name link entityType owners { uid name } }`;
+            } );
+
+          /* Executing the queries */
+          const resolvedSources = await GqlHelper.execSimpleQuery( { queries, fragments: [ GqlHelper.fragments.homeType ] } )
+            .then( res => res.data );
+
+          /* Merging the query output with the NotificationsConfig output */
+          return notificationConfigs
+            .map( ( notificationConfig, index ) => {
+              notificationConfig.source = resolvedSources[ `source_${ index }` ][ 0 ];
+              return notificationConfig;
+            } );
+        } ).catch( err => {
+          throw err;
+        } );
+    },
+    getNotificationsConfigByID ( root: any, { id }: GraphQLArgs, ctx: any ) {
+      return NotificationsConfig
+        .findById( id )
+        .lean<NotificationsConfigType>()
+        .exec();
+    }
+  },
+  /*NotificationsConfigType: {
+    source: (root: any, args: any, ctx: any) => {
+      return ["New" + root.source];
+    },
+  },*/
+  Mutation: {
+    createNotificationsConfig ( root: any, { notificationsConfig }: GraphQLArgs, ctx: any ) {
+      const data = new NotificationsConfig( notificationsConfig );
+      return data.save();
+    },
+    updateNotificationsConfig ( root: any, { notificationsConfig }: GraphQLArgs, ctx: any ) {
+      /* Optional: if you want to send graphql subscription updates when this query is called) */
+      // pubsub.publish(Notifications_UPDATE, data);
+      return NotificationsConfig
+        .findByIdAndUpdate( notificationsConfig.id, notificationsConfig, { new: true } )
+        .exec();
+    },
+    deleteNotificationsConfig ( root: any, args: GraphQLArgs, ctx: any ) {
+      return NotificationsConfig.findByIdAndRemove( args.id ).exec();
+    },
+  },
+  // Subscription: {
+  //   createdNotificationsConfig: {
+  //     subscribe: () => pubsub.asyncIterator( NOTIFICATIONS_CONFIG_CREATE )
+  //   },
+  //   updatedNotificationsConfig: {
+  //     subscribe: () => pubsub.asyncIterator( NOTIFICATIONS_CONFIG_UPDATE )
```

Comment:
This is a wrong way of triggering the update

Reply:
@riginoommen Yes, that's why it is commented. I think that's the default output of the CLI.
It is a work in progress, and a part of another issue.